in_toto/in_toto_record.py

```diff
@@ -14,166 +13,141 @@
+  key_args_group.add_argument("-k", "--key", type=str,
+      help="Path to private key to sign link metadata (PEM)",
+      metavar="<signing key path>")
+  key_args_group.add_argument("-g", "--gpg", nargs="?", const=True,
+      metavar="<gpg keyid>", help=("GPG keyid to sign link metadata "
+      "(if set without argument, the default key is used)"))
+
+  parent_parser.add_argument("--gpg-home", dest="gpg_home", type=str,
+      help="Path to GPG keyring (if not set the default keyring is used)",
+      metavar="<gpg keyring path>")
 
-  in_toto_args.add_argument("-k", "--key", type=str, required=True,
-      help="Path to private key to sign link metadata (PEM)")
+  parent_parser.add_argument("-v", "--verbose", dest="verbose",
+      help="Verbose execution.", default=False, action="store_true")
 
-  in_toto_args.add_argument("-v", "--verbose", dest='verbose',
-      help="Verbose execution.", default=False, action='store_true')
+  subparser_start = subparsers.add_parser("start", parents=[parent_parser])
+  subparser_stop = subparsers.add_parser("stop", parents=[parent_parser])
 
   subparser_start.add_argument("-m", "--materials", type=str, required=False,
-      nargs='+', help="Files to record before link command execution")
+      nargs='+', help="Files to record before link command execution",
+      metavar="<material path>")
 
   subparser_stop.add_argument("-p", "--products", type=str, required=False,
-      nargs='+', help="Files to record after link command execution")
+      nargs='+', help="Files to record after link command execution",
+      metavar="<product path>")
 
   args = parser.parse_args()
 
   # Turn on all the `log.info()` in the library
   if args.verbose:
-    log.logging.getLogger().setLevel(log.logging.INFO)
+    in_toto.log.logging.getLogger().setLevel(in_toto.log.logging.INFO)
 
   # Override defaults in settings.py with environment variables and RCfiles
   in_toto.user_settings.set_settings()
 
+  # If `--gpg` was set without argument it has the value `True` and
+  # we will try to sign with the default key
+  gpg_use_default = (args.gpg == True)
+
+  # Otherwise we interpret it as actual keyid
+  gpg_keyid = None
+  if args.gpg != True:
+    gpg_keyid = args.gpg
+
   # We load the key here because it might prompt the user for a password in
   # case the key is encrypted. Something that should not happen in the library.
+  key = None
+  if args.key:
+    try:
+      key = in_toto.util.prompt_import_rsa_key_from_file(args.key)
+
+    except Exception as e:
+      in_toto.log.error("in load key - {}".format(e))
+      sys.exit(1)
+
   try:
-    key = in_toto.util.prompt_import_rsa_key_from_file(args.key)
+    if args.command == "start":
+      in_toto.runlib.in_toto_record_start(args.step_name, args.materials,
+          signing_key=key, gpg_keyid=gpg_keyid,
+          gpg_use_default=gpg_use_default, gpg_home=args.gpg_home)
+
+    elif args.command == "stop": # pragma: no branch
```

Comment:
I could go both ways, but if argparse already catches args.command == "start" || "stop" we could just use an else here no?

Reply:
I had the same thoughts (in various places throughout the repo). I am not sure about *the right way* to do this. For the sake of readability something like `else: # args.command == "stop"` might be a good compromise. :)

Reply:
FYI https://github.com/secure-systems-lab/code-style-guidelines/issues/18